### PR TITLE
Allow creating empty tours & renaming tours

### LINKED ISSUE #3713: I want to be able to rename a tour/route after it has been created
**In order to**  `achieve something`
Have the ability to rename a route from the admin panel as a dispatcher or administrator

### LINKED ISSUE #3715: Add "+" button so user can add a tour in the tour panel + add a delete action on created tours
Have action icons on the tours panel like in the unassigned panel

- [ ]  create tour button (a plus)
- [ ]  other stuff?

current visual
image

unassigned modal with action icons
image

### LINKED ISSUE #3717: When right-clicking on several tasks I want to be able to create a tour (no pickup mandatory)
**Describe the bug**
when selecting various tasks in the admin panel as a dispatcher or admin the way it does or does not allow the creation of routes is confusing.

No route option or any options- highlighting various pickups and no dropoff
image

Route option - highlighting dropoff and pickups can make a route
image

Group option but no route - highlighting various pickups
image

### LINKED ISSUE #3818: Hitting twice the "create tour" button submit twice (creates two tours)
**Expected behavior**
The button in the modal must be locked when hit once 

**Screenshots**

### PR DESCRIPTION
**Pull request by jimakker**
More like *first-class citizen* Tours initial phase.

Missing features:
* rename Tour
* delete Tour

Fixes #3713
Fixes #3715
Fixes #3818
Fixes #3717

### CODE REVIEW

**Comment by jimakker** [comment]
This test is failing, don't know if related https://github.com/coopcycle/coopcycle-web/pull/3775/files#diff-fb6c1883e2728a96550614899bee97662c9d8c8fdced6e490394ba6620045310L462

**Comment by Atala** [comment]
Related issues
#3717 
#3713 
#3818 
#3715 
coopcycle/coopcycle-app#1608

File: app/DoctrineMigrations/Version20230807105004.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+final class Version20230807105004 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql('ALTER TABLE tour ADD date DATE');
```
Comment (alexsegura) [inline app/DoctrineMigrations/Version20230807105004.php:22]: From what I see when running `make migrations-diff`, an index should be added on this column. 

```php
$this->addSql('CREATE INDEX IDX_6AD1F969AA9E377A ON tour (date)');
```

File: app/DoctrineMigrations/Version20230807105004.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        $this->addSql('COMMENT ON COLUMN task_list.date IS NULL');
```
Comment (alexsegura) [inline app/DoctrineMigrations/Version20230807105004.php:23]: Does not seem to be needed, does nothing.

File: app/DoctrineMigrations/Version20230807105004.php (hunk 1 above)
Comment (alexsegura) [inline app/DoctrineMigrations/Version20230807105004.php:22]: Should be: 

```
$this->addSql('ALTER TABLE tour ADD date DATE NOT NULL');
```

In this case, for the migration to work, you would need to first add it as DEFAULT NULL, then update data, and finally change it to NOT NULL, like in [this migration](https://github.com/coopcycle/coopcycle-web/blob/4b894429515216b79a4b9fa2dea7bcd4047d3192/app/DoctrineMigrations/Version20231025114558.php).

**Review by Atala — APPROVED** [review]

**Review by alexsegura — COMMENTED** [review]
When a task is removed from a tour, it appears both in the tour panel and in the unassigned tasks panel until it's saved

tour_unassign_disappear

**Review by alexsegura — APPROVED** [review]

**Pull request merged**